### ISSUE TITLE
Baseline CapsNet model doesn't work.

### ISSUE THREAD

**Issue by TolerantChief** [moejoe95/MoCapsNet#4]
Hi. I've tried using the baseline Capsule Networks model but it doesn't get past 4% of accuracy, no matter how many epochs it's been training for. And loss is nan.

In the following example I was using a custom dataset but I got the same result with CIFAR and MNIST.

`######## PYTORCH MODEL BUILT ########
Num params: 13690624
######## RUN STARTED ########
MODE;EPOCH;LOSS:ACCURACY;TIME
TRAIN;1;nan;0.042097488921713444;56.694s
TEST;1;nan;0.03723404255319149;8.887s
TRAIN;2;nan;0.042097488921713444;36.131s
TEST;2;nan;0.03723404255319149;7.723s
TRAIN;3;nan;0.042097488921713444;23.574s
TEST;3;nan;0.03723404255319149;3.308s
TRAIN;4;nan;0.042097488921713444;22.103s
TEST;4;nan;0.03723404255319149;3.157s
TRAIN;5;nan;0.042097488921713444;22.152s
TEST;5;nan;0.03723404255319149;3.159s
TRAIN;6;nan;0.042097488921713444;22.371s
TEST;6;nan;0.03723404255319149;3.23s
TRAIN;7;nan;0.042097488921713444;22.399s
TEST;7;nan;0.03723404255319149;3.251s
TRAIN;8;nan;0.042097488921713444;21.971s
TEST;8;nan;0.03723404255319149;3.873s
TRAIN;9;nan;0.042097488921713444;21.674s
TEST;9;nan;0.03723404255319149;4.077s
TRAIN;10;nan;0.042097488921713444;21.707s
TEST;10;nan;0.03723404255319149;3.619s
TRAIN;11;nan;0.042097488921713444;21.812s`

**Comment by moejoe95** [comment]
Hi,
have you tried tweaking the learning rate? Maybe try to decrease it by a
factor of 10.


TolerantChief ***@***.***> schrieb am Do., 13. Apr. 2023,
18:42:

**Comment by TolerantChief** [comment]
Hi,
Sorry for taking so long to respond.
I've tried increasing and decreasing the learning rate and I still have the same problem. The odd thing is that I don't have that issue with Daniel Havir's model, but I can't get my custom dataset to work on that one.

**Comment by moejoe95** [comment]
can you give the exact command with arguments you are using?

TolerantChief ***@***.***> schrieb am Di., 9. Mai 2023, 18:36:

**Comment by TolerantChief** [comment]
I'm using the model in Google Colab and this is the command that I'm using:
`!python /content/MoCapsNet/main.py --data_path '/content/data/' --dataset cifar --epochs 30`

**Comment by moejoe95** [comment]
you need to set the parameter `residual` or `momentum`, without one of them the model is not able to learn anything. Maybe take a look at our paper on archive: https://arxiv.org/abs/2201.11091 where we explain everything in detail :)

**Comment by TolerantChief** [comment]
But if use the `residual` and `momentum` parameters I will use either the `ResCapsNet` model or the `MoCapsNet` model right? I've used those ones and they work perfectly, but when I try to use the baseline `CapsNet` model is when I get that issue.

**Comment by moejoe95** [comment]
The baseline CapsNet does not train because it is deeper (more capsule layers) than the original model. That is exactly what we wanted to show in our research: if we want to be able to train deeper capsule networks, we need to use residual layers.
